arkouda/pdarrayIO.py

```diff
@@ -633,15 +633,20 @@ def save_all(
             datasetNames = [str(column) for column in range(len(columns))]
     if (mode.lower() not in "append") and (mode.lower() not in "truncate"):
         raise ValueError("Allowed modes are 'truncate' and 'append'")
-    first_iter = True
-    for arr, name in zip(pdarrays, cast(List[str], datasetNames)):
-        """Append all pdarrays to existing files as new datasets EXCEPT the first one,
-        and only if user requests truncation"""
-        if mode.lower() != "append" and first_iter:
-            arr.save(prefix_path=prefix_path, dataset=name, file_format=file_format, mode="truncate")
-            first_iter = False
-        else:
+
+    if mode.lower() == "append":
+        for arr, name in zip(pdarrays, cast(List[str], datasetNames)):
             arr.save(prefix_path=prefix_path, dataset=name, file_format=file_format, mode="append")
+    else:
+        first_iter = True
+        for arr, name in zip(pdarrays, cast(List[str], datasetNames)):
+            """Append all pdarrays to existing files as new datasets EXCEPT the first one,
+            and only if user requests truncation"""
+            if first_iter:
+                arr.save(prefix_path=prefix_path, dataset=name, file_format=file_format, mode="truncate")
+                first_iter = False
+            else:
+                arr.save(prefix_path=prefix_path, dataset=name, file_format=file_format, mode="append")
```

Comment:
I am not sure this hits the mark for what we want. It is adding an additional loop that instead of 1. I would recommend something like 

```python
    for arr, name in zip(pdarrays, cast(List[str], datasetNames)):
        arr.save(prefix_path=prefix_path, dataset=name, file_format=file_format, mode=mode)
        if mode.lower() == "truncate":
            mode = "append"
```

If we wanted a simple fix, we could change line 640 from master to 
```python
if mode.lower() == "truncate" and first_iter
```

Overall, this does not look like it is as much of an issue as initially thought. I would suggest going with one of the solutions above to address the issue.

Side note: I forgot to click add when I did my review initially, sorry!

Reply:
I like your first suggestion.